x-pack/filebeat/fbreceiver/receiver_test.go

```diff
@@ -412,3 +416,102 @@ func getFromSocket(t *testing.T, sb *strings.Builder, socketPath string, endpoin
 
 	return true
 }
+
+type logGenerator struct {
+	t           *testing.T
+	tmpDir      string
+	f           *os.File
+	filePattern string
+	sequenceNum int64
+}
+
+func newLogGenerator(t *testing.T, tmpDir string) *logGenerator {
+	return &logGenerator{
+		t:           t,
+		tmpDir:      tmpDir,
+		filePattern: "input-*.log",
+	}
+}
+
+func (g *logGenerator) Start() {
+	f, err := os.CreateTemp(g.tmpDir, g.filePattern)
+	require.NoError(g.t, err)
+	g.f = f
+}
+
+func (g *logGenerator) Stop() {
+	require.NoError(g.t, g.f.Close())
+	require.NoError(g.t, os.Remove(g.f.Name()))
```

Comment:
I think we have the potential for a race condition on Windows if the "reader" still has the file open the remove will fail on Windows.  Probably better to have the `testing.T` pass in a tempdir it has made and it will remove at the end of the test.  That has logic to try and deal with the Windows issues.

https://cs.opensource.google/go/go/+/refs/tags/go1.24.5:src/testing/testing.go;l=1290-1318

Reply:
Thanks, I removed the call to os.Remove. This file is already created inside `t.TmpDir` so it should be removed at the end of the test.